nixos/modules/programs/singularity.nix

```diff
@@ -3,32 +3,83 @@
 with lib;
 let
   cfg = config.programs.singularity;
-  singularity = pkgs.singularity.overrideAttrs (attrs : {
-    installPhase = attrs.installPhase + ''
-      mv $out/libexec/singularity/bin/starter-suid $out/libexec/singularity/bin/starter-suid.orig
-      ln -s /run/wrappers/bin/singularity-suid $out/libexec/singularity/bin/starter-suid
-    '';
-  });
-in {
+in
+{
   options.programs.singularity = {
-    enable = mkEnableOption "Singularity";
+    enable = mkEnableOption "singularity" // {
+      description = ''
+        Whether to install Singularity/Apptainer
+        and add SUID support.
+      '';
+    };
+    package = mkOption {
+      default = pkgs.singularity;
+      defaultText = "pkgs.singularity";
+      description = ''
+        Singularity package to override and install. Typical values are
+        pkgs.singularity, pkgs.singularity-legacy, pkgs.apptainer, and pkgs.singularity-ce.
+      '';
+      type = types.package;
+    };
+    packageOverriden = mkOption {
+      default = mkDefault null;
+      defaultText = "lib.mkDefault null";
+      description = ''
+        The overriden singularity package to install.
+        <note>
+          <para>
+            This option is handled internally to place the overriden result of
+            <varname>programs.singularity.package</varname> with SUID support added.
+          </para>
+          <para>
+            Specify with <varname>lib.mkForce</varname> if you want to do it yourself.
+          </para>
+        </note>
+      '';
+      type = types.nullOr types.package;
+    };
+    projectName = mkOption {
```

Comment:
> Update: It just come to me that I place this option to deal with the possibility that packageOverriden could be forcefully overrided with a package of different projectName

Why not use `cfg.packageOverriden.projectName` still?

Reply:
Because `packageOverriden` depends on `package.projectName`...

I got it! I'll use `packageOverriden.projectName` on other stuff while using `package.projectName` to make `packageOverriden`.

Reply:
Applied and removed.